cmd/generate/main.go

```diff
@@ -8,58 +8,130 @@
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"log"
 	"os"
 
 	"github.com/nginxinc/nginx-go-crossplane/internal/generator"
 )
 
+type cmdConfig struct {
```

Comment:
I'm not sure why you need what looks like a duplicate of the `generator.GenerateConfig`. I think you can just have one type.

Reply:
At our former discussion, https://github.com/nginxinc/nginx-go-crossplane/pull/112#discussion_r1693504383
We want to parse `filter` as an array first, then transfer it into a map.

Reply:
I don't think you need another type to do this. The linked thread includes the [example I provided](https://go.dev/play/p/qcrjzlIPGHc) to accomplish this. You can decode the JSON with the array into the map type without an intermediate, duplicate struct.

Reply:
Oh I see. Sorry I misunderstood the example before. Applied.